spec/routing/project_routing_spec.rb

```diff
@@ -60,25 +60,34 @@
+  describe 'miscellaneous' do
+    it { expect(get('/projects/123/settings')).to     route_to(controller: 'projects', action: 'settings', id: '123') }
+    it { expect(put('projects/123/modules')).to       route_to(controller: 'projects', action: 'modules', id: '123') }
+    it { expect(put('projects/123/custom_fields')).to route_to(controller: 'projects', action: 'custom_fields', id: '123') }
+    it { expect(put('projects/123/archive')).to       route_to(controller: 'projects', action: 'archive', id: '123') }
+    it { expect(put('projects/123/unarchive')).to     route_to(controller: 'projects', action: 'unarchive', id: '123') }
+    it { expect(post('projects/123/copy')).to         route_to(controller: 'copy_projects', action: 'copy', id: '123') }
     it { expect(get('projects/123/copy_project_from_settings')).to route_to(controller: 'copy_projects', action: 'copy_project', id: '123', coming_from: 'settings') }
-    it { expect(post('projects/123/copy')).to route_to(controller: 'copy_projects', action: 'copy',         id: '123') }
+  end
+
+  describe 'settings' do
+    it { expect(get('/projects/123/settings/info')).to          route_to(controller: 'projects', action: 'settings', id: '123', tab: 'info') }
+    it { expect(get('/projects/123/settings/modules')).to       route_to(controller: 'projects', action: 'settings', id: '123', tab: 'modules') }
+    it { expect(get('/projects/123/settings/members')).to       route_to(controller: 'projects', action: 'settings', id: '123', tab: 'members') }
+    it { expect(get('/projects/123/settings/custom_fields')).to route_to(controller: 'projects', action: 'settings', id: '123', tab: 'custom_fields') }
+    it { expect(get('/projects/123/settings/versions')).to      route_to(controller: 'projects', action: 'settings', id: '123', tab: 'versions') }
+    it { expect(get('/projects/123/settings/categories')).to    route_to(controller: 'projects', action: 'settings', id: '123', tab: 'categories') }
+    it { expect(get('/projects/123/settings/repositories')).to  route_to(controller: 'projects', action: 'settings', id: '123', tab: 'repositories') }
+    it { expect(get('/projects/123/settings/boards')).to        route_to(controller: 'projects', action: 'settings', id: '123', tab: 'boards') }
+    it { expect(get('/projects/123/settings/activities')).to    route_to(controller: 'projects', action: 'settings', id: '123', tab: 'activities') }
+    it { expect(get('/projects/123/settings/types')).to         route_to(controller: 'projects', action: 'settings', id: '123', tab: 'types') }
   end
 
   describe 'types' do
-    it do
-      expect(patch('/projects/123/types')).to route_to(controller: 'projects',
-                                                       action: 'types',
-                                                       id: '123')
-    end
+    it { expect(patch('/projects/123/types')).to route_to(controller: 'projects', action: 'types', id: '123') }
   end
```

Comment:
I ignore them on purpose as I want to take care of all / the most in one rush. ok with that? will do as soon as rails 4.1 is merged.